db/view/build_progress_virtual_reader.hh

```diff
@@ -140,6 +152,8 @@ class build_progress_virtual_reader {
                                 adjust_ckey(scylla_in_progress_rt.end),
                                 scylla_in_progress_rt.end_kind,
                                 scylla_in_progress_rt.tomb);
+                    } else if (mf.is_end_of_partition()) {
+                        _previous_clustering_key = clustering_key::make_empty();
```

Comment:
I think it's safer to use `std::optional<clustering_key>` for this, because clustering keys may be prefixes only really for compact storage tables, so creating an empty clustering_key for this schema is not really correct. I know it happens to work today, but still.

Reply:
done